src/opm/parser/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.cpp

```diff
@@ -107,7 +107,7 @@ namespace {
                                         "Use either SGOF (or SLGOF) and/or SWOF or SGFN/SWFN and SOF2/SOF3");
         }
 
-        if (!family1 && !family2) {
+        if ((! (family1 || family2)) && (twoP || threeP)) {
```

Comment:
Personally I find these multi condition logical expression hard to read and prefer to split them up. I realize this is slightly differently semantically - but maybe it is also correct - or even better:
```C++
if (twoP || threeP) {
    if (family1)
       return SatfuncFamily::I
    if (familiy2)
       return SatfuncFamily::II;
    throw std::invalid_argument(...);
} else
   return SatfuncFamiliy::none;
```

Reply:
> Personally I find these multi condition logical expression hard to read and prefer to split them up.

I have the same problem with nested conditionals...  In any case I've pushed an update inspired by your suggestion.

Reply:
Hehe - sone like the mother and some like the daughter!